skbio/io/tests/test_util.py

```diff
@@ -634,6 +638,27 @@ def test_open_write_mode(self):
         fh.close()
         self.assertTrue(fh.closed)
 
+class TestReadStandardInput(unittest.TestCase):
+    num_text = ["ONE", "TWO", "THREE", "FOUR", "FIVE", "SIX", "SEVEN"]
+
+    def test_stdin_read_fasta(self):
```

Comment:
These tests should stay, but I recommend additionally setting up an independent integration test run through the `Makefile`. Something like the below. The inline python though could also be placed in a minor script. 

```bash
#!/bin/bash

cat skbio/io/tests/test_data/some_fasta.fna | \
    python -c "import skbio, sys; print([r.metadata['id'] for r in skbio.read(sys.stdin, format='fasta')]) > /tmp/obs_stdin.ids

grep "^>" skbio/io/tests/test_data/some_fasta.fna | tr -d ">" > /tmp/exp_stdin.ids

obs_md5=$(md5sum /tmp/obs_stdin.ids)
exp_md5=$(md5sum /tmp/exp_stdin.ids)

if [[ ${obs_md5} != ${exp_md5} ]]; then
    exit 1
fi
```

Reply:
I'm not super familiar with these, but I've been looking for a way to do just this! I'll take a look at implementing this for testing.